lib/strategies/webapp-strategy.js

```diff
@@ -344,12 +345,20 @@ function getTokens(req, options, strategy) {
 		code: req.query.code
 	};
 
+	function cb(err, user, info) {
+		if (err) {
+			return strategy.fail;
+		} else {
+			return strategy.success(user || null, info);
+		}
+	}
 	retrieveTokens(formData, strategy).then(function (appIdAuthContext) {
 		// Save authorization context to HTTP session
 		req.session[WebAppStrategy.AUTH_CONTEXT] = appIdAuthContext;
+		let user = appIdAuthContext.identityTokenPayload.sub;
 		delete req.session[WebAppStrategy.STATE_PARAMETER];
 		logger.debug("completeAuthorizationFlow :: success", options);
-		strategy.success(appIdAuthContext.identityTokenPayload || null);
```

Comment:
We changed the behavior here a bit. Notice how here we call success with the whole `IDTokenPayload`, but in cb we call success with `user`, which contains only the `sub` field of the IDToken. We need to make sure to still call success with the whole IDToken